http-server-netty/src/main/java/io/micronaut/http/server/netty/NettyHttpRequest.java

```diff
@@ -164,6 +166,12 @@ public class NettyHttpRequest<T> extends AbstractNettyHttpRequest<T> implements
 
     private final BodyConvertor bodyConvertor = newBodyConvertor();
 
+    /**
+     * ONLY for NettyBodyAnnotationBinder use.
+     */
+    @Internal
+    public ArgumentBinder.BindingResult<ConvertibleValues<?>> convertibleBody;
+
```

Comment:
I was thinking we could have a custom object representing the body, which can accept some-kind of visitor, so we can avoid `instanceof` for cases when the body is known.